napari/layers/points/points.py

```
@@ -232,15 +305,65 @@ def __init__(
         self._is_selecting = False
         self._clipboard = {}
 
-        self.edge_color = normalize_and_broadcast_colors(
-            len(self.data), self._current_edge_color
-        )
-        self.face_color = normalize_and_broadcast_colors(
-            len(self.data), self._current_face_color
-        )
+        # set the edge color properties
+        self._edge_color_property = ''
+        if self._is_color_mapped(edge_color):
+            self._edge_color_mode = ColorMode.CYCLE
+        else:
+            self._edge_color_mode = ColorMode.DIRECT
+        if edge_color_cycle is None:
+            self._edge_color_cycle = cycle(
+                np.array([[1, 1, 0, 1], [0, 1, 0, 1]])
+            )
+        elif isinstance(edge_color_cycle, cycle):
+            self._edge_color_cycle = edge_color_cycle
+        else:
```

Maybe add a comment here explaining what this catch-all clause is? It's actually the most common case, I would imagine...

Actually, I would say, would it be possible to modify `transform_color_cycle` so that you can unify this clause and the one above it, ie, if it's already a cycle, you just return it. That way, it becomes an easy case: if it's none, use this simple cycle, otherwise, use what was provided while making sure it's a cycle.

---

Good point. I have modified `transform_color_cycle` to allow `cycle` objects to pass through.